crates/types/src/traits/block_contents.rs

```diff
@@ -43,6 +43,13 @@ pub trait Transaction:
 {
 }
 
+/// Abstraction over the minimum block size for transaction submitted to the builder
+pub trait BuilderTransaction {
+    /// The function to estimate transaction size
+    /// It takes in the transaction itself plus additional_length_overhead which includes transaction table header, transaction table entry etc.
+    fn minimum_block_size(&self, additional_length_overhead: u64) -> u64;
```

Comment:
Pasting from Zulip:
> Why are we passing additional_length_overhead as a parameter to the trait method? It depends exclusively on the concrete Transaction type implementing the trait, not on the caller. In fact, caller has no idea what the overhead is without knowing concrete NodeTypes. The whole ((NS_ID_BYTE_LEN + NS_OFFSET_BYTE_LEN) + NUM_TXS_BYTE_LEN + TX_OFFSET_BYTE_LEN) formula applies only to SeqTypes, because that's how the sequencer's BlockHeader works; for TestTypes, for example, the overhead is just 0, because TestTypes::BlockHeader is empty

Reply:
Removed the parameter `additional_length_overhead`